PowerShellTools/DebugEngine/ScriptProgramProvider.cs

```diff
@@ -23,24 +27,17 @@ public class ScriptProgramProvider : IDebugProgramProvider2
         public int GetProviderProcessData(enum_PROVIDER_FLAGS Flags, IDebugDefaultPort2 pPort, AD_PROCESS_ID ProcessId, CONST_GUID_ARRAY EngineFilter, PROVIDER_PROCESS_DATA[] pProcess)
         {
             Log.Debug("ProgramProvider: GetProviderProcessData");
-            return VSConstants.E_NOTIMPL;
-
-#if FALSE
-
-            if (Flags.HasFlag(enum_PROVIDER_FLAGS.PFLAG_GET_PROGRAM_NODES))
+            try
             {
-                var process = Process.GetProcessById((int) ProcessId.dwProcessId);
-                foreach (ProcessModule module in process.Modules)
+                if (Flags.HasFlag(enum_PROVIDER_FLAGS.PFLAG_GET_PROGRAM_NODES))
                 {
-                    if (module.ModuleName.StartsWith("System.Management.Automation", StringComparison.OrdinalIgnoreCase))
+                    if(PowerShellToolsPackage.DebuggingService.IsAttachable(ProcessId.dwProcessId))

```

Comment:
Again, for x86 pshost attaching to x64 process, you give false from this check, so it would miss a bunch of work done inside this "if", is the scenario still working then?